public/app/features/alerting/unified/components/rule-editor/notificaton-preview/NotificationPreview.tsx

```diff
@@ -0,0 +1,147 @@
+  customLabels: Array<{
+    key: string;
+    value: string;
+  }>;
+  alertQueries: AlertQuery[];
+  condition: string;
+  folder: Folder;
+  alertName?: string;
+  alertUid?: string;
+}
+
+export const NOTIFICATION_PREVIEW_TITLE = 'Alert instance routing preview';
+
+export const NotificationPreview = ({
+  alertQueries,
+  customLabels,
+  condition,
+  folder,
+  alertName,
+  alertUid,
+}: NotificationPreviewProps) => {
+  const styles = useStyles2(getStyles);
+  // potential instances are the instances that are going to be routed to the notification policies
+  //const [potentialInstances, setPotentialInstances] = useState<AlertInstances[]>([]);
+
+  const { usePreviewMutation } = alertRuleApi;
+
+  const [trigger, { data = [], isLoading, isUninitialized: previewUninitialized }] = usePreviewMutation();
+  // any time data is updated from trigger, we need to update the potential instances
+  //  convert data to list of labels: are the representation of the potential instances
+  const potentialInstances = compact(data.flatMap((label) => label?.labels));
+
+  const onPreview = () => {
+    // Get the potential labels given the alert queries, the condition and the custom labels (autogenerated labels are calculated on the BE side)
+    trigger({
+      alertQueries: alertQueries,
+      condition: condition,
+      customLabels: customLabels,
+      folder: folder,
+      alertName: alertName,
+      alertUid: alertUid,
+    });
+  };
+
+  // Get list of alert managers source name + image
+  const alertManagerSourceNamesAndImage = useGetAlertManagersSourceNamesAndImage();
+
+  const onlyOneAM = alertManagerSourceNamesAndImage.length === 1;
+  const renderHowToPreview = !Boolean(data?.length) && !isLoading;
+
+  return (
+    <Stack direction="column" gap={2}>
+      <div className={styles.routePreviewHeaderRow}>
+        <h4 className={styles.previewHeader}>{NOTIFICATION_PREVIEW_TITLE}</h4>
+        <div className={styles.button}>
+          <Button icon="sync" variant="secondary" type="button" onClick={onPreview}>
+            Preview routing
+          </Button>
+        </div>
+      </div>
+      {!renderHowToPreview && (
+        <div className={styles.textMuted}>
+          Based on the labels you have added above and the labels that have been automatically assigned, alert instances
+          are being route to notification policies in the way listed bellow. Expand the notification policies to see the
+          instances which are going to be routed to them.
+        </div>
+      )}
+      {isLoading && <div className={styles.textMuted}>Loading...</div>}
+      {renderHowToPreview && (
+        <div className={styles.previewHowToText}>
+          {`When your query and labels are configured, click "Preview routing" to see the results here.`}
+        </div>
+      )}
+      {!isLoading && !previewUninitialized && potentialInstances.length > 0 && (
+        <Suspense fallback={<LoadingPlaceholder text="Loading preview..." />}>
```

Comment:
First 'loading indicator' is about getting the potential alert instances once user clicks the preview button (is about getting the response). 
On the other hand, the fallback is for the `NotificationPreviewByAlertManager` being rendered as it contains some async code that can take some time. And we wait until we have the list of potential instances as we need them to call the algorithm that calculate the matching policies. As this process can take some time, we wrap this into a `Supense` with a callback that says: `Loading preview`. I don't find a case where this loading can be indefinitely shown, if something fails in the `NotificationPreviewByAlertManager` a message of error is shown, instead of the loading placeholder.